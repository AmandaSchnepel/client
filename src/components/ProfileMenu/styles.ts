import styled, { css, DefaultTheme } from 'styled-components'

export const Nav = styled.nav`
  ${({ theme }) => css`
    flex-direction: column;
    border: 0;
    a:not(:last-child) {
      border-bottom: 0.1rem solid ${theme.colors.grayLine};
    }
  `}
`
const linkModifiers = {
  default: (theme: DefaultTheme) => css`
    background: ${theme.colors.white1};
    color: ${theme.colors.black1};
  `,

  active: (theme: DefaultTheme) => css`
    background: ${theme.colors.green};
    color: ${theme.colors.white1};
  `
}

type LinkProps = {
  isActive?: boolean
}

export const Link = styled.a<LinkProps>`
  ${({ theme, isActive }) => css`
    text-decoration: none;
    display: flex;
    align-items: center;
    padding: ${theme.spacings.xsmall} ${theme.spacings.small};
    transition: background, color, ${theme.transition.default};
    &:hover {
      background: ${theme.colors.green};
      color: ${theme.colors.white1};
    }
    > span {
      margin-left: ${theme.spacings.xsmall};
    }
    ${!isActive && linkModifiers.default(theme)};
    ${isActive && linkModifiers.active(theme)};
  `}
`
