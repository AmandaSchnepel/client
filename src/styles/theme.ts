export default {
  grid: {
    container: '130rem',
    gutter: '3.2rem'
  },
  border: {
    radius: '0.4rem'
  },
  font: {
    family:
      "Poppins, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    light: 300,
    regular: 400,
    medium: 500,
    bold: 700
  },
  sizes: {
    xsmall: '1.2rem',
    small: '1.4rem',
    medium: '1.6rem',
    icons: '1.5rem',
    large: '1.8rem',
    xlarge: '2.0rem',
    xxlarge: '2.4rem',
    infos: '2.6rem'
  },
  colors: {
    green: '#207824',
    lightgreen: '#90BC92',
    black1: '#212529',
    black2: '#030517',
    white1: '#FFFFFF',
    white2: '#FAFAFA',
    blackgray: '#434343',
    gray: '#31322D',
    grayLine: '#D6D6D6',
    grayBox: '#939393',
    lightGray1: '#767676',
    lightGray2: '#EAEAEA',
    grayConter: '#9A9A9A',
    red: '#EF2F2F',
    lightred: '#ED8383',
    marrom: '#FFC690'
  },
  spacings: {
    xxsmall: '0.55rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  },
  transition: {
    default: '0.3s ease-in-out',
    fast: '0.1s ease-in-out'
  }
}
