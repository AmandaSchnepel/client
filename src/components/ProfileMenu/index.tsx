//ICON endereço
import { MapMarkedAlt as AddressIcon } from '@styled-icons/fa-solid/MapMarkedAlt'
//ICON MEIOS DE PAGAMENTO
import { Payments as PaymentIcon } from '@styled-icons/material-outlined/Payments'
// ICON MEUS PEDIDOS
import { ShoppingBags as RequestIcon } from '@styled-icons/boxicons-solid/ShoppingBags'
// icon alterar senha
import { LockPassword as PasswordIcon } from '@styled-icons/remix-fill/LockPassword'
// ICON DE NOSSAS LOJAS
import { Storefront as StoreIcon } from '@styled-icons/material/Storefront'
//icon minha conta
import { AccountCircle, ExitToApp } from '@styled-icons/material-outlined'

import Link from 'next/link'

import * as S from './styles'

export type ProfileMenuProps = {
  activeLink?:
    | '/profile/me'
    | '/profile/endereços'
    | '/profile/pagamentos'
    | '/profile/pedidos'
    | '/profile/senha'
    | '/profile/lojas'
}

const ProfileMenu = ({ activeLink }: ProfileMenuProps) => (
  <S.Nav>
    <Link href="/profile/me" passHref>
      <S.Link isActive={activeLink === '/profile/me'} title="Meu perfil">
        <AccountCircle size={24} />
        <span>Meu perfil</span>
      </S.Link>
    </Link>

    <Link href="/profile/endereços" passHref>
      <S.Link isActive={activeLink === '/profile/endereços'} title="Endereços">
        <AddressIcon size={24} />
        <span>Endereços</span>
      </S.Link>
    </Link>

    <Link href="/profile/pagamentos" passHref>
      <S.Link
        isActive={activeLink === '/profile/pagamentos'}
        title="Meios de pagamentos"
      >
        <PaymentIcon size={24} />
        <span>Meios de pagamento</span>
      </S.Link>
    </Link>

    <Link href="/profile/pedidos" passHref>
      <S.Link isActive={activeLink === '/profile/pedidos'} title="Pedidos">
        <RequestIcon size={24} />
        <span>Pedidos</span>
      </S.Link>
    </Link>

    <Link href="/profile/senha" passHref>
      <S.Link isActive={activeLink === '/profile/senha'} title="Alterar Senha">
        <PasswordIcon size={24} />
        <span>Alterar Senha</span>
      </S.Link>
    </Link>

    <Link href="/profile/lojas" passHref>
      <S.Link isActive={activeLink === '/profile/lojas'} title="Nossas Lojas">
        <StoreIcon size={24} />
        <span>Nossas Lojas</span>
      </S.Link>
    </Link>

    <Link href="/logout" passHref>
      <S.Link>
        <ExitToApp size={24} />
        <span>Sign out</span>
      </S.Link>
    </Link>
  </S.Nav>
)

export default ProfileMenu
