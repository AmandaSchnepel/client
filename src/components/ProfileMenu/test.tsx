import { screen } from '@testing-library/react'
import theme from 'styles/theme'
import { renderWithTheme } from 'utils/tests/helpers'
import ProfileMenu from '.'

describe('<ProfileMenu />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<ProfileMenu />)

    expect(
      screen.getByRole('link', { name: /Meu perfil/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('link', { name: /Enderecos/i })).toBeInTheDocument()
    expect(
      screen.getByRole('link', { name: /Meios de pagamento/i })
    ).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /Pedidos/i })).toBeInTheDocument()
    expect(
      screen.getByRole('link', { name: /Alterar Senha/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('link', { name: /Nossas Lojas/i })
    ).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /Sign out/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render the menu with an active link defined', () => {
    renderWithTheme(<ProfileMenu activeLink="/perfil/enderecos" />)

    expect(screen.getByRole('link', { name: /Enderecos/i })).toHaveStyle({
      background: theme.colors.green,
      color: theme.colors.white1
    })
  })
})
