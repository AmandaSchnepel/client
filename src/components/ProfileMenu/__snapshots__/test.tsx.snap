// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileMenu /> should render the heading 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 0;
}

.c0 a:not(:last-child) {
  border-bottom: 0.1rem solid #D6D6D6;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1.6rem 2.4rem;
  -webkit-transition: background,color,0.3s ease-in-out;
  transition: background,color,0.3s ease-in-out;
  background: #FFFFFF;
  color: #212529;
}

.c1:hover {
  background: #207824;
  color: #FFFFFF;
}

.c1 > span {
  margin-left: 1.6rem;
}

<nav
  class="c0"
>
  <a
    class="c1"
    href="/profile/me"
    title="Meu perfil"
  >
    <svg
      aria-hidden="true"
      class="c2"
      fill="currentColor"
      focusable="false"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33A7.95 7.95 0 014 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83zM12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
      />
    </svg>
    <span>
      Meu perfil
    </span>
  </a>
  <a
    class="c1"
    href="/profile/endere%C3%A7os"
    title="Endereços"
  >
    <svg
      aria-hidden="true"
      class="c2"
      fill="currentColor"
      focusable="false"
      height="24"
      viewBox="0 0 576 512"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M288 0c-69.59 0-126 56.41-126 126 0 56.26 82.35 158.8 113.9 196.02 6.39 7.54 17.82 7.54 24.2 0C331.65 284.8 414 182.26 414 126 414 56.41 357.59 0 288 0zm0 168c-23.2 0-42-18.8-42-42s18.8-42 42-42 42 18.8 42 42-18.8 42-42 42zM20.12 215.95A32.006 32.006 0 0 0 0 245.66v250.32c0 11.32 11.43 19.06 21.94 14.86L160 448V214.92c-8.84-15.98-16.07-31.54-21.25-46.42L20.12 215.95zM288 359.67c-14.07 0-27.38-6.18-36.51-16.96-19.66-23.2-40.57-49.62-59.49-76.72v182l192 64V266c-18.92 27.09-39.82 53.52-59.49 76.72-9.13 10.77-22.44 16.95-36.51 16.95zm266.06-198.51L416 224v288l139.88-55.95A31.996 31.996 0 0 0 576 426.34V176.02c0-11.32-11.43-19.06-21.94-14.86z"
        fill="currentColor"
      />
    </svg>
    <span>
      Endereços
    </span>
  </a>
  <a
    class="c1"
    href="/profile/pagamentos"
    title="Meios de pagamentos"
  >
    <svg
      aria-hidden="true"
      class="c2"
      fill="currentColor"
      focusable="false"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="none"
        height="24"
        width="24"
      />
      <path
        d="M19 14V6c0-1.1-.9-2-2-2H3c-1.1 0-2 .9-2 2v8c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2zm-2 0H3V6h14v8zm-7-7c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3zm13 0v11c0 1.1-.9 2-2 2H4v-2h17V7h2z"
      />
    </svg>
    <span>
      Meios de pagamento
    </span>
  </a>
  <a
    class="c1"
    href="/profile/pedidos"
    title="Pedidos"
  >
    <svg
      aria-hidden="true"
      class="c2"
      fill="currentColor"
      focusable="false"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 20h2V10a1 1 0 0 1 1-1h12V7a1 1 0 0 0-1-1h-3.051c-.252-2.244-2.139-4-4.449-4S6.303 3.756 6.051 6H3a1 1 0 0 0-1 1v11a2 2 0 0 0 2 2zm6.5-16c1.207 0 2.218.86 2.45 2h-4.9c.232-1.14 1.243-2 2.45-2z"
      />
      <path
        d="M21 11H9a1 1 0 0 0-1 1v8a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2v-8a1 1 0 0 0-1-1zm-6 7c-2.757 0-5-2.243-5-5h2c0 1.654 1.346 3 3 3s3-1.346 3-3h2c0 2.757-2.243 5-5 5z"
      />
    </svg>
    <span>
      Pedidos
    </span>
  </a>
  <a
    class="c1"
    href="/profile/senha"
    title="Alterar Senha"
  >
    <svg
      aria-hidden="true"
      class="c2"
      fill="currentColor"
      focusable="false"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M18 8h2a1 1 0 011 1v12a1 1 0 01-1 1H4a1 1 0 01-1-1V9a1 1 0 011-1h2V7a6 6 0 1112 0v1zm-2 0V7a4 4 0 10-8 0v1h8zm-5 6v2h2v-2h-2zm-4 0v2h2v-2H7zm8 0v2h2v-2h-2z"
      />
    </svg>
    <span>
      Alterar Senha
    </span>
  </a>
  <a
    class="c1"
    href="/profile/lojas"
    title="Nossas Lojas"
  >
    <svg
      aria-hidden="true"
      class="c2"
      fill="currentColor"
      focusable="false"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="none"
        height="24"
        width="24"
      />
      <path
        d="M21.9 8.89l-1.05-4.37c-.22-.9-1-1.52-1.91-1.52H5.05c-.9 0-1.69.63-1.9 1.52L2.1 8.89c-.24 1.02-.02 2.06.62 2.88.08.11.19.19.28.29V19c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-6.94c.09-.09.2-.18.28-.28.64-.82.87-1.87.62-2.89zm-2.99-3.9l1.05 4.37c.1.42.01.84-.25 1.17-.14.18-.44.47-.94.47-.61 0-1.14-.49-1.21-1.14L16.98 5l1.93-.01zM13 5h1.96l.54 4.52c.05.39-.07.78-.33 1.07-.22.26-.54.41-.95.41-.67 0-1.22-.59-1.22-1.31V5zM8.49 9.52L9.04 5H11v4.69c0 .72-.55 1.31-1.29 1.31-.34 0-.65-.15-.89-.41a1.42 1.42 0 01-.33-1.07zm-4.45-.16L5.05 5h1.97l-.58 4.86c-.08.65-.6 1.14-1.21 1.14-.49 0-.8-.29-.93-.47-.27-.32-.36-.75-.26-1.17zM5 19v-6.03c.08.01.15.03.23.03.87 0 1.66-.36 2.24-.95.6.6 1.4.95 2.31.95.87 0 1.65-.36 2.23-.93.59.57 1.39.93 2.29.93.84 0 1.64-.35 2.24-.95.58.59 1.37.95 2.24.95.08 0 .15-.02.23-.03V19H5z"
      />
    </svg>
    <span>
      Nossas Lojas
    </span>
  </a>
  <a
    class="c1"
    href="/logout"
  >
    <svg
      aria-hidden="true"
      class="c2"
      fill="currentColor"
      focusable="false"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5a2 2 0 00-2 2v4h2V5h14v14H5v-4H3v4a2 2 0 002 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
      />
    </svg>
    <span>
      Sign out
    </span>
  </a>
</nav>
`
